src/content/plugins/normal-module-replacement-plugin.md

```diff
@@ -40,50 +41,78 @@ Conditional build depending on an [specified environment](/configuration/configu
 Say you want a configuration with specific values for different build targets.
 
 ``` javascript
-module.exports = function(env) {
-  var appTarget = env.APP_TARGET || 'VERSION_A';
+module.exports = function(env, args) {
   return {
     plugins: [
       new webpack.NormalModuleReplacementPlugin(/(.*)-APP_TARGET(\.*)/, function(resource) {
-        resource.request = resource.request.replace(/-APP_TARGET/, `-${appTarget}`);
+        var original = resource.request;
+        var replace = resource.request.replace(/-APP_TARGET/, `-${env}`);
+        resource.request = replace;
+        // This will help you debug the replacement during the build process.
+        console.log(`Original Config: '${original}', Replacement: '${replace}'`);
       })
     ]
   };
 
 };
 ```
 
-Create the two config files:
+Create the two config files (the conventions are the same for JavaScript and TypeScript):
 
-__app/config-VERSION_A.js__
+__config/config-BETA.js (OR .ts)__ 
 
 ``` javascript
 export default {
-  title : 'I am version A'
+  title : 'I am the Beta configuration.'
 };
 ```
 
-__app/config-VERSION_B.js__
+__config/config-PROD.js (OR .ts)__
 
 ``` javascript
 export default {
-  title : 'I am version B'
+  title : 'I am the Production configuration.'
 };
 ```
 
 Then import that config using the keyword you're looking for in the regexp:
 
 ``` javascript
-import config from 'app/config-APP_TARGET';
+import config from './config/config-APP_TARGET';
 console.log(config.title);
 ```
 
-And now you just get the right config imported depending on which target you're building for:
+And now the right config will be imported depending on which target you're building for:
 
 ``` shell
-webpack --env.APP_TARGET VERSION_A
-=> 'I am version A'
+webpack --mode production --env BETA
+=> 'I am the Beta configuration.'
 
-webpack --env.APP_TARGET VERSION_B
-=> 'I am version B'
+webpack --mode production --env PROD
+=> 'I am the Production configuration.'
 ```
+
+
+## TypeScript Considerations
+
+The TypeScript compiler requires that all files be physically-present in order to generate an output. If you attempt to
+complete the above tasks, you will likely receive the following error:
```

Comment:
Suggest changing to: After running the above example in TypeScript, you would likely receive the following error:

Reply:
Well, you're not "running it in TypeScript". TypeScript is a compiler not a runtime engine. The problem happens when you run Webpack and ts-loader tries to finish its work compiling the code during that process. I can change to "After making the above changes and running Webpack, `ts-loader` is likely to throw the following error:"

Reply:
yup much better, pls do

Reply:
I've made the change as requested.